pkg/nmi/server/server.go

```diff
@@ -125,6 +132,49 @@ func (fn appHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	latency := time.Since(start)
 	logger.Infof("Status (%d) took %d ns", rw.statusCode, latency.Nanoseconds())
 }
+func (s *Server) hostHandler(logger *log.Entry, w http.ResponseWriter, r *http.Request) {
+	hostIP := parseRemoteAddr(r.RemoteAddr)
+	 if hostIP != "127.0.0.1" {
+		msg := "request remote address is not from a host"
+		logger.Error(msg)
+		http.Error(w, msg, http.StatusInternalServerError)
+		return
+	 }
+
+	 podns, podname := parseRequstHeader(r)
+	 if podns == "" || podname == "" {
+		 logger.Errorf("Error getting podname and podns from request")
+		 http.Error(w, "Error getting podname and podns from request", http.StatusInternalServerError)
+		 return
+	 }
+
+	 podIDs, err := s.KubeClient.ListPodIds(podns, podname)
+	 if err != nil || len(*podIDs) == 0 {
+		 msg := fmt.Sprintf("no AzureAssignedIdentity found for pod:%s/%s", podns, podname)
+		 logger.Errorf("%s, %+v", msg, err)
+		 http.Error(w, msg, http.StatusForbidden)
```

Comment:
we can extract this as a function that can be used by both handlers ? to reduce code duplication